HCollections/HListT.fs

```diff
@@ -6,11 +6,11 @@ open TypeEquality
 [<NoEquality>]
 type HListT<'ts, 'elem> =
     private
-    | Empty of Teq<'ts, unit>
-    | Cons of HListTConsCrate<'ts, 'elem> * 'ts TypeList
+    | Empty of 'ts HList * Teq<'ts, unit>
```

Comment:
I wouldn't bother storing the HList here in the empty case, you can always magic one up later

Reply:
Sure, I just did so because it made some of the code (toTypeList and toHList) slightly nicer in my opinion. Happy to change though